src/flightRecorder.cpp

```diff
@@ -1074,6 +1074,7 @@ class Recording {
         buf->putVar32(tid);
         buf->putVar32(call_trace_id);
         buf->putVar32(event->_thread_state);
+        buf->putUtf8(event->_ecid);
```

Comment:
This will not work with long strings (>120 characters).
As mentioned in [the comment](https://github.com/jvm-profiling-tools/async-profiler/pull/576#issuecomment-1092314951), it would be better to have a numeric context ID instead.

Reply:
I believe you are right, for now >50% of overhead of ```setEcid``` is done by ```jni_GetStringUTFChars```, so that overhead can be eliminated.

Reply:
done